crypto/fipsmodule/sha/sha3.c

```diff
@@ -35,7 +71,19 @@ int SHA3_Init(KECCAK1600_CTX *ctx, uint8_t pad, size_t bit_len) {
         exit(1);
   }
 
-  size_t block_size = SHA3_BLOCKSIZE(bit_len);
+  size_t block_size;
+
+  // The block size in SHA3 depends on the digest |bit_len|
+  // The block size in SHAKE depends only on the security level
+  // The |SHA3_PAD_CHAR| differs for SHA3 and SHAKE
+  // It is used to assign the block size based on the 
+  // digest length (for SHA3) or the security level for (SHAKE)
+  if (pad == SHA3_PAD_CHAR) {
+    block_size = SHA3_BLOCKSIZE(bit_len);
+  } else {
```

Comment:
what if `pad` is some value that's different from both `SHA3_PAD_CHAR` and `SHAKE_PAD_CHAR`?

Reply:
I added `return 0` case since a wrong result will be produced if not. Sorry for this.